src/librustdoc/clean/mod.rs

```diff
@@ -1927,7 +1940,34 @@ impl Clean<Type> for hir::Ty {
                     _ => Infer // shouldn't happen
                 }
             }
-            TyBareFn(ref barefn) => BareFunction(box barefn.clean(cx)),
```

Comment:
`BareFnTy` has its own `Clean` impl, you should be changing that, not this - this is why I didn't want you to change the enum variant, but the struct.